### ISSUE TITLE
LAM diagnostic tool implementation for HAT 

### ISSUE THREAD

**Issue by oubotong** [XPixelGroup/HAT#62]
The paper mentions that HAT uses LAM tool to picture the area of pixels that contribute to the image reconstruction task.
However, the original LAM paper only publishes a demo of their tool. I am wondering if HAT can open-source their implementation with LAM as well.

Thanks!

**Comment by oubotong** [comment]
I am trying to port HAT to LAM with [hat_arch.py](https://github.com/XPixelGroup/HAT/blob/main/hat/archs/hat_arch.py), which I think defines the network, but I got the following results when using LAM:
![fa038d881e105cd45a671bca6ebf721](https://user-images.githubusercontent.com/26356127/232962143-a2416a65-39a4-489a-a6ff-5cdc1e874985.png)

I don't know if I am missing any part for example the [hat_model.py](https://github.com/XPixelGroup/HAT/blob/main/hat/models/hat_model.py) file. I really appreciate it if you can help me with this.

**Comment by ChamingHong** [comment]
> I am trying to port HAT to LAM with [hat_arch.py](https://github.com/XPixelGroup/HAT/blob/main/hat/archs/hat_arch.py), which I think defines the network, but I got the following results when using LAM: ![fa038d881e105cd45a671bca6ebf721](https://user-images.githubusercontent.com/26356127/232962143-a2416a65-39a4-489a-a6ff-5cdc1e874985.png)
> 
> I don't know if I am missing any part for example the [hat_model.py](https://github.com/XPixelGroup/HAT/blob/main/hat/models/hat_model.py) file. I really appreciate it if you can help me with this.

Hello, I used the LAM-demo code and got the same results as you. Have you solved this problem?

**Comment by yang-yee** [comment]
Yes, I also used the hat_arch. py file and the weights of HAT_SRx4, and obtained the same result. I feel that the receptive field used is not that large, and I don't know where the error occurred. I don't know how the author implemented the LAM diagram? Have you solved it yet?
